public/enhanced-table-vis-controller.js

```diff
@@ -308,12 +308,30 @@ module.controller('EnhancedTableVisController', function ($scope, Private, confi
 
       let removedCounter = 0;
       _.forEach(hiddenColumns, function (item) {
-        let index = getRealColIndex(parseInt(item), splitColIndex);
-        table.columns.splice(index - removedCounter, 1);
-        _.forEach(table.rows, function (row) {
-          row.splice(index - removedCounter, 1);
-        });
-        removedCounter++;
+        try {
+          if (parseInt(item).toString() == item) {
```

Comment:
Here, could you use : `isNaN(parseInt(item))`

Reply:
I actually can't because if someone has a column named "1-whatever column", because of how JS works, it's not gonna return a NaN if you convert into an integer. It's gonna return 1.
That's why I had to do that horrible stuff. Should you have another way to accomplish such a task let me know

Reply:
OK.
So another way:
`if (item.matches(/ *[0-9]+ */) {`

Reply:
Regex looks good. We're talking of elasticsearch after all ;)
Though, I think I'll leave it in a dedicate function `isInt()` to make the code more readable.

Reply:
Nice.